packages/build/src/core/dev.js

```diff
@@ -0,0 +1,140 @@
+import { handleBuildError } from '../error/handle.js'
+import { logTimer, logBuildSuccess } from '../log/messages/core.js'
+import { trackBuildComplete } from '../telemetry/main.js'
+import { reportTimers } from '../time/report.js'
+
+import { execBuild, startBuild } from './build.js'
+import { getSeverity } from './severity.js'
+
+export const startDev = async (devCommand, flags = {}) => {
```

Comment:
`devCommand` is a callback that executes the dev command. In the future, we might want to create a core step that is responsible for figuring out this command based on data like framework detection, but for now we'll rely on consumers (like CLI) to supply it.